crates/apollo-compiler/src/validation/mod.rs

```diff
@@ -39,6 +44,230 @@ pub trait ValidationDatabase:
     fn validate_object(&self) -> Vec<ApolloDiagnostic>;
     fn validate_operation(&self) -> Vec<ApolloDiagnostic>;
     fn validate_unused_variable(&self) -> Vec<ApolloDiagnostic>;
+
+    fn check_directive_definition(&self, directive: hir::DirectiveDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_object_type_definition(&self, object_type: hir::ObjectTypeDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_interface_type_definition(&self, object_type: hir::InterfaceTypeDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_union_type_definition(&self, object_type: hir::UnionTypeDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_enum_type_definition(&self, object_type: hir::EnumTypeDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_input_object_type_definition(&self, object_type: hir::InputObjectTypeDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_schema_definition(&self, object_type: hir::SchemaDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_selection_set(&self, selection_set: hir::SelectionSet) -> Vec<ApolloDiagnostic>;
+    fn check_arguments_definition(&self, arguments_def: hir::ArgumentsDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_field_definition(&self, field: hir::FieldDefinition) -> Vec<ApolloDiagnostic>;
+    fn check_input_values(&self, input_values: Arc<Vec<hir::InputValueDefinition>>) -> Vec<ApolloDiagnostic>;
+    fn check_db_definitions(&self, definitions: Arc<Vec<hir::Definition>>) -> Vec<ApolloDiagnostic>;
+    fn check_directive(&self, schema: hir::Directive) -> Vec<ApolloDiagnostic>;
+    fn check_arguments(&self, schema: Vec<hir::Argument>) -> Vec<ApolloDiagnostic>;
+    fn check_field(&self, field: Arc<hir::Field>) -> Vec<ApolloDiagnostic>;
```

Comment:
what would be the semantic difference of `validate_` and `check_`? Should we try merging them?

Reply:
Ah, nothing, it's just to prevent naming collisions while migrating.

Reply:
Gotcha!